test cases/common/106 generatorcustom/meson.build

```diff
@@ -2,12 +2,17 @@ project('generatorcustom', 'c')
 
 creator = find_program('gen.py')
 catter = find_program('catter.py')
+gen_res3 = find_program('gen-res3.py')
 
 gen = generator(creator,
   output: '@BASENAME@.h',
   arguments : ['@INPUT@', '@OUTPUT@'])
 
-hs = gen.process('res1.txt', 'res2.txt')
+res3 = custom_target('gen-res3',
+    output : 'res3.txt',
+    command : [gen_res3, '@OUTPUT@'])
+
+hs = gen.process('res1.txt', 'res2.txt', res3)
```

Comment:
You should add a test for a `CustomTargetIndex` as well, something like `hs2 = gen.process('res1.txt', 'res2.txt', res3[0])` is probably sufficient.